website/docs/sdk-reference/dotnet.md

```diff
@@ -240,19 +240,66 @@ User userObject = new User("john@example.com");
 {
     Email = "john@example.com",
     Country = "United Kingdom",
-    Custom = new Dictionary<string, string> {
-        {"SubscriptionType", "Pro"},
-        {"UserRole", "Admin"}}
+    Custom =
+    {
+        ["SubscriptionType"] = "Pro",
+        ["UserRole"] = "Admin"
+    }
+};
+```
+
+The `Custom` dictionary also allows attribute values other than `string` values:
+
+```csharp
+User userObject = new User("#UNIQUE-USER-IDENTIFIER#")
+{
+    Custom =
+    {
+        ["Rating"] = 4.5,
+        ["RegisteredAt"] = DateTimeOffset.Parse("2023-11-22 12:34:56 +00:00", CultureInfo.InvariantCulture),
+        ["Roles"] = new[] { "Role1", "Role2" }
+    }
 };
 ```
 
+The set of allowed attribute values depends on the comparison type of the condition which references the User Object attribute.
+`string` values are supported by all comparison types (in some cases they need to be provided in a specific format though).
+Some of the comparison types work with other types of values, as described below.
+
+**Text-based comparisons** (EQUALS, IS ONE OF, etc.)
+* accept `string` values,
+* all other values are automatically converted to `string` (a warning will be logged but evaluation will continue as normal).
+
+**SemVer-based comparisons** (IS ONE OF, &lt;, &gt;=, etc.)
+* accept `string` values containing a properly formatted, valid semver value,
+* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).
+
+**Number-based comparisons** (=, &lt;, &gt;=, etc.)
+* accept `double` values and all other numeric values which can safely be converted to `double`,
+* accept `string` values containing a properly formatted, valid `double` value,
+* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).
+  
+**Date time-based comparisons** (BEFORE / AFTER - available only in the new config model, *Config V2*)
+* accept `DateTime` or `DateTimeOffset` values, which are automatically converted to a second-based Unix timestamp,
+* accept `double` values representing a second-based Unix timestamp and all other numeric values which can safely be converted to `double`,
+* accept `string` values containing a properly formatted, valid `double` value,
+* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).
+  
+**String array-based comparisons** (ARRAY CONTAINS ANY OF / ARRAY NOT CONTAINS ANY OF - available only in the new config model, *Config V2*)
+* accept arrays of `string`,
+* accept `string` values containing a valid JSON string which can be deserialized to an array of `string`,
+* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).
+
+It is also worth mentioning that in case a non-string attribute value needs to be converted to `string` during evaluation,
```

Comment:
I don't get what's the point here.

Who needs to convert? Is it going to be converted automatically?

Reply:
Yep, this is about an automatic conversion and "during evaluation" are the key words here.

As we agreed recently, we drop the concept of User Object attribute helpers and allow users to pass non-string custom attribute values to the evaluation logic.

When a non-string custom attribute is used as the base of percentage options evaluation, we need to convert that attribute value to string (since it's included in the value fed to the hash function which calculates the hash buckets).